java/src/main/java/ai/rapids/cudf/DType.java

```diff
@@ -16,77 +16,229 @@
+  public static final DType FLOAT32 = new DType(DTypeEnum.FLOAT32);
+  public static final DType FLOAT64 = new DType(DTypeEnum.FLOAT64);
+  public static final DType BOOL8 = new DType(DTypeEnum.BOOL8);
+  public static final DType TIMESTAMP_DAYS = new DType(DTypeEnum.TIMESTAMP_DAYS);
+  public static final DType TIMESTAMP_SECONDS = new DType(DTypeEnum.TIMESTAMP_SECONDS);
+  public static final DType TIMESTAMP_MILLISECONDS = new DType(DTypeEnum.TIMESTAMP_MILLISECONDS);
+  public static final DType TIMESTAMP_MICROSECONDS = new DType(DTypeEnum.TIMESTAMP_MICROSECONDS);
+  public static final DType TIMESTAMP_NANOSECONDS = new DType(DTypeEnum.TIMESTAMP_NANOSECONDS);
+  public static final DType DURATION_DAYS = new DType(DTypeEnum.DURATION_DAYS);
+  public static final DType DURATION_SECONDS = new DType(DTypeEnum.DURATION_SECONDS);
+  public static final DType DURATION_MILLISECONDS = new DType(DTypeEnum.DURATION_MILLISECONDS);
+  public static final DType DURATION_MICROSECONDS = new DType(DTypeEnum.DURATION_MICROSECONDS);
+  public static final DType DURATION_NANOSECONDS = new DType(DTypeEnum.DURATION_NANOSECONDS);
+  public static final DType STRING = new DType(DTypeEnum.STRING);
+  public static final DType LIST = new DType(DTypeEnum.LIST);
+  public static final DType STRUCT = new DType(DTypeEnum.STRUCT);
+
+  private static final DType[] staticDTypes = new DType[]{
+      EMPTY,
+      INT8,
+      INT16,
+      INT32,
+      INT64,
+      UINT8,
+      UINT16,
+      UINT32,
+      UINT64,
+      FLOAT32,
+      FLOAT64,
+      BOOL8,
+      TIMESTAMP_DAYS,
+      TIMESTAMP_SECONDS,
+      TIMESTAMP_MILLISECONDS,
+      TIMESTAMP_MICROSECONDS,
+      TIMESTAMP_NANOSECONDS,
+      DURATION_DAYS,
+      DURATION_SECONDS,
+      DURATION_MILLISECONDS,
+      DURATION_MICROSECONDS,
+      DURATION_NANOSECONDS,
+      null, // DICTIONARY32
+      STRING,
+      LIST,
+      null, // DECIMAL32
+      null, // DECIMAL64
+      STRUCT
+  };
+
   /**
-   * s since the UNIX epoch
+   * This only works for fixed width types. Variable width types like strings the value is
+   * undefined and should be ignored.
+   *
+   * @return size of type in bytes.
    */
-  TIMESTAMP_SECONDS(8, 13, "timestamp[s]"),
+  public int getSizeInBytes() { return typeId.sizeInBytes; }
+
   /**
-   * ms since the UNIX epoch
+   * Returns scale for Decimal Type
+   * @return scale
    */
-  TIMESTAMP_MILLISECONDS(8, 14, "timestamp[ms]"),
+  public int getScale() { return scale; }
+
+  public String getSimpleName() { return typeId.simpleName; }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    DType type = (DType) o;
+    return scale == type.scale && typeId == type.typeId;
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(typeId, scale);
+  }
+
+  @Override
+  public String toString() {
+    return "" + typeId;
```

Comment:
This should be updated to print the scale iff its a decimal type

Reply:
I am not sure if this is the good way to print for decimal types. Any suggestions are welcome.